### PR TITLE
Dev

### LINKED ISSUE #73: v-for image not found 404 using nuxt-image
If I loop through an array with v-for the images are not found with a 404 error
`<div v-for="image in product[0].images" v-bind:key="image.rank">`
`    <nuxt-image :src="require(`~/assets/images/shop/${image.image_570xN}`)" />`
 `</div>`

but if I just use the img tag, the image is found
`<div v-for="image in product[0].images" v-bind:key="image.rank">`
`    `
 `</div>`

Please can someone suggest a solution?

### LINKED ISSUE #80: Placeholder image is not hidden after main image loads
Nuxt config:

```javascript
export default {
    target: 'static',a
    buildModules: ['@nuxtjs/tailwindcss'],
    modules: ['@nuxt/image',],
    image: {
        providers: {
            cloudinary: {
                baseURL: 'https://res.cloudinary.com/milos5593/image/upload/',
            },
        },
    },
}
```

And then in the component:

```vue
<template>
    <header class="bg-black">
        <div class="p-4 max-w-screen-xl">
            <!-- Logo -->
                <nuxt-link
                    class="font-bold text-lg text-white focus:outline-none focus:shadow-outline"
                    to="/"
                >
                <nuxt-image
                    class="w-56 h-auto"
                    fit="contain"
                    :placeholder="true"
                    :src="'cloudinary:/crossroads_logo.png'"
                    no-script
                />
            </nuxt-link>
        </div>
    </header>
</template>

<script>
export default {
    name: 'SiteHeader',
}
</script>
```

### LINKED ISSUE #81: [dx] top level provider options
```js
export default {
 image: {
    cloudinary: {}
 }
 random: {
   provider: '~~/providers/random',
   options: {

   }
 }
}
```

### LINKED ISSUE #82: [docs] dedicated section/page for each provider

### LINKED ISSUE #83: [dx] rename ìpx`back to local
or alias..

### LINKED ISSUE #84: support `NUXT_IMAGE_PROVIDER` environment variable
to override options.provider` (we may also auto detect it based on deployment target)

### LINKED ISSUE #85: <nuxt-img> alias

### LINKED ISSUE #86: <nuxt-image> rename responsive to layout
`<NuxtImage :responsive="true">` ~> `<NuxtImage layout="responsive">`

### LINKED ISSUE #88: Remove blur filter on custom placeholders

### LINKED ISSUE #92: Module injects placeholder's base64 in payload even when placeholder not used
If you use `NuxtImage` without `width` & `height`, like 

```html
<NuxtImage src="cloudinary:/vue-telemetry/4d02deda597adab13fac9341c22458aa.jpg" />
``` 

it will inject placeholder base64 data into payload. 

Kapture 2020-11-13 at 13 23 21

If you add `width` & `height` like 

```html
<NuxtImage width="4" height="3" src="cloudinary:/vue-telemetry/4d02deda597adab13fac9341c22458aa.jpg" />
```

placeholder disappears from payload. 

Reproduction: https://github.com/bdrtsky/image-module-issues

### LINKED ISSUE #93: srcset is not generating correct image urls
I am using module with Cloudinary, and have setup like:

```js
image: {
    presets: [
      {
        name: '4/3',
        modifiers: {
          fit: 'cover',
          format: 'auto',
        }
      }
    ],
    providers: {
      cloudinary: {
        baseURL: 'https://res.cloudinary.com/nuxt/image/upload/'
      }
    }
  },
```

```html
<NuxtImage sizes="100,300:300,500:500,1000:1000" responsive src="cloudinary+4/3:/vue-telemetry/4d02deda597adab13fac9341c22458aa.jpg" />
```

I expect module to generate srcset with different size image variations, but got same image url for each viewport:

```4d02deda597adab13fac9341c22458aa```

Reproduction: https://github.com/bdrtsky/image-module-issues

### LINKED ISSUE #97: Non default base path breaks nuxt image
When using a non-default base path (i.e. `/`) then the components stops working.

If this is a configuration issue, would it please be possible to describe in the docs how to modify the setup to work with non-default base paths?

Thanks.

### LINKED ISSUE #102: 400 (Bad Request)
Hello, I just installed nuxt image but when I try to load image from static folder I get an error:

```
GET http://localhost:3000/_image/ipx/local/_/_/the_eye_slow.gif 400 (Bad Request)
```

My component:

```
<template>
  <div @click="countClick()">
    <UiSpacer top />

    <nuxt-image
      :src="`/the_eye_${state}.gif`"
      alt="oeil du jeu"
      :width="imgDimention"
      :height="imgDimention"
    />
  </div>
</template>

<script>
import getTailwindBps from '@/mixins/getTailwindBps.js'

export default {
  name: 'UiEye',

  components: {
    UiSpacer: require('@/components/ui/UiSpacer.vue').default,
  },

  mixins: [getTailwindBps],

  props: {
    state: {
      type: String,
      required: true,
      validator(value) {
        return ['focus', 'normal', 'slow']
      },
    },
  },
```

My image is stored in the root of static folder. Can someone help me ?

### PR DESCRIPTION
**Pull request by farnabaz**
- Simplify NuxtImage functionality. Remove the placeholder and lazy feature.
- Remove unnecessary tags and only render image tag
- Rename responsive prop to layout (fix #86)
- Introduce quality prop
- rename `nuxt-image` to `nuxt-img` (fix #85)
- rename ipx to local (close #83)
- clean up double slashes outside of provider runtime
- handle the router base (fix #97)
- support NUXT_IMAGE_PROVIDER environment variable (close #84)
- allow top-level providers config (fix #81)
    official providers config can be defined at top-level of image config and custom providers should define under providers property
- refactor `nuxt-picture` (fix #92)
- disable blur filter on custom placeholder (fix #88)
- merge presets with inline modifiers (fix #93)
- support assets using the remote adapter (fix #73)
- introduce `provider` and `preset` props
- support GIF files and data URLs (fix #102)

close #80 
close #82

### CODE REVIEW

File: src/runtime/nuxt-img.ts [hunk 1]
```diff
@@ -83,15 +83,10 @@ export default {
...
-    // const { width, height } = await this.$img.getResolution(this.src, {
-    //   provider: this.provider,
-    //   preset: this.preset
-    // })
-    const { width, height } = await this.$img.getMeta(this.src, {
+    const { width, height } = await this.$img.getResolution(this.src, {
       provider: this.provider,
-      preset: this.preset
+      preset: this.preset,
+      modifiers: this.modifiers
```
Comment (Atinux) [inline src/runtime/nuxt-img.ts:89]: Are the modifiers and preset required?

Comment (farnabaz) [reply src/runtime/nuxt-img.ts:89]: Modifiers may change the size of the image, like `rotate`

**Comment by NotHolst** [comment]
When are these changes expected be released

**Comment by florentb** [comment]
We've got minor changes to add regarding TwicPics provider. Should we do the pull request into dev branch or wait for this merge to be done?

**Comment by Atinux** [comment]
@florentb I think you can make a PR to dev branch

File: docs/content/en/custom-provider.md [hunk 2]
```diff
@@ -123,11 +115,11 @@ export default {
...
 
-### Use provider in `nuxt-image`
-To use your provider, write your provider's name before the src. Read [provider doc](/nuxt-image#provider)
+### Use provider in `nuxt-img`
+To use your provider, write your provider's name before the src. Read [provider doc](/nuxt-img#provider)
 
 ```vue{}[index.vue]
 <template>
-  <nuxt-image src="custom:/main.png" />
+  <nuxt-img provider="customProvider src="/main.png" />
```
Comment (TheLearneer) [inline docs/content/en/custom-provider.md:123]:
```suggestion
  <nuxt-img provider="customProvider" src="/main.png" />
```

File: docs/content/en/examples/random-provider.md [hunk 3]
```diff
@@ -58,15 +58,15 @@ Let's use our random provider in our Nuxt application:
 
   ```vue{}[~/pages/index.js]
   <template>
-    <nuxt-image src="random:/" width="600" height="400" />
+    <nuxt-img provider="random"random src="/" width="600" height="400" />
```
Comment (TheLearneer) [inline docs/content/en/examples/random-provider.md:61]:
```suggestion
    <nuxt-img provider="random" src="/" width="600" height="400" />
```

I guess that is not supposed to be there

File: docs/content/en/nuxt-img.md [hunk 4]
```diff
@@ -11,110 +11,50 @@ Path to image file. `src` sould be in form of absolute path and starts with `/`.
...
+Specify width of the image.  
+`nuxt-img` uses this value to optimize the image as best as its possible:
 
-<!-- TODO: multiple providers -->
-<!-- TODO: default provider -->
+- This value will set as `width` attribute of DOM element
+- On `layout != "responsive"`
+  - When both `width` and `height` are present and values isn't equal to `auto`, this value will use to resize the image.
+  - On the server-side rendering when the value equals to `auto` and height is not set or equal to `auto`, `nuxt-img` will automatically detects image's size and use it. (This not happens on client-side rendering)
+  - On the server-side rendering when the value equals to `auto` and `height` is set, `nuxt-img` will automatically detects image's size and calulates proper with based on image's aspect ratio and `height` value. (This not happens on client-side rendering)
```
Comment (TheLearneer) [inline docs/content/en/nuxt-img.md:28]:
```suggestion
  - On the server-side rendering when the value equals to `auto` and `height` is set, `nuxt-img` will automatically detects image's size and calulates proper width based on image's aspect ratio and `height` value. (This not happens on client-side rendering)
```

Comment (TheLearneer) [inline docs/content/en/nuxt-img.md:28]:
```suggestion
  - On the server-side rendering when the value equals to `auto` and `height` is set, `nuxt-img` will automatically detects image's size and calulates proper width based on image's aspect ratio and `height` value. (This not happens on client-side rendering)
```
Isn't that supposed to be width?

File: docs/content/en/nuxt-img.md [hunk 5]
```diff
@@ -11,110 +11,50 @@ Path to image file. `src` sould be in form of absolute path and starts with `/`.
...
+Specify height of the image.  
+`nuxt-img` uses this value to optimize the image as best as its possible:
 
-<code-group>
-  <code-block label="index.vue" active>
+- This value will set as `height` attribute of DOM element
+- On `layout != "responsive"`
+  - When both `width` and `height` are present and values isn't equal to `auto`, this value will use to resize the image.
+  - On the server-side rendering when the value equals to `auto` and width is not set or equal to `auto`, `nuxt-img` will automatically detects image's size and use it. (This not happens on client-side rendering)
+  - On the server-side rendering when the value equals to `auto` and `width` is set, `nuxt-img` will automatically detects image's size and calulates proper with based on image's aspect ratio and `width` value. (This not happens on client-side rendering)
```
Comment (TheLearneer) [inline docs/content/en/nuxt-img.md:41]:
```suggestion
  - On the server-side rendering when the value equals to `auto` and `width` is set, `nuxt-img` will automatically detects image's size and calulates proper height based on image's aspect ratio and `width` value. (This not happens on client-side rendering)
```
Just like above, isn't it supposed to be height?

File: docs/content/en/nuxt-picture.md [hunk 6]
```diff
@@ -7,67 +7,98 @@ category: Componets
...
       }
     }
   }
-  </script>
   ```
+  </code-block>
+  <code-block label="Preview">
+
+  <div class="text-center p-4 bg-gray-800 rounded-b-md">
+    <nuxt-img lazy="false" provider="cloudinary" preset="jpg-cover" src="/remote/nuxt-org/blog/going-full-static/main.png" />
```
Comment (TheLearneer) [inline docs/content/en/nuxt-picture.md:60]:
```suggestion
    <nuxt-img :lazy="false" provider="cloudinary" preset="jpg-cover" src="/remote/nuxt-org/blog/going-full-static/main.png" />
```

File: docs/content/en/providers/local.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+category: Providers
+---
+
+
+Internally nuxt image uses [ipx](https://github.com/nuxt-contrib/ipx) to modify and optimize images.  
+
+- `dir`: The root directory of the all images. By default nuxt image looks `static` dir to find original images, 
+- `clearCache`: The ipx has a caching stategy to clear cached images to reduce massive disk usages. You can schedule the cache cleaning job using `clearCache` option in provide options. By default this cron job is disabled.
+- `cacheDir`: The directory to store the cached images.
+- `sharp`: Modify default behavior of image opetimizer. Note that if you change this option, you need to clear ipx cache.
```
Comment (TheLearneer) [inline docs/content/en/providers/local.md:14]:
```suggestion
- `sharp`: Modify default behavior of image optimizer. Note that if you change this option, you need to clear ipx cache.
```

**Comment by TheLearneer** [comment]
Any updates on this ??

**Comment by NotHolst** [comment]
Did this PR die, or is there something we can do to help ?

**Comment by pi0** [comment]
Main branch changed to `dev` for time being. Next release will be shortly after finishing last part of refactors (runtime/components) and doc update.

**Pull request closed without merging**